src/main/java/htsjdk/samtools/cram/io/ITF8.java

```diff
@@ -150,37 +174,62 @@ public static int readUnsignedITF8(final ByteBuffer buffer) {
      * @param value the value to be written out
      * @param buffer   the {@link ByteBuffer} to write to
      */
-    public static void writeUnsignedITF8(final int value, final ByteBuffer buffer) {
+    public static int writeUnsignedITF8(final int value, final ByteBuffer buffer) {
         if ((value >>> 7) == 0) {
+            // no control bits
             buffer.put((byte) value);
-            return;
+            return 8;
         }
 
         if ((value >>> 14) == 0) {
-            buffer.put((byte) ((value >> 8) | 128));
+            // 1 control bit
+            buffer.put((byte) ((value >> 8) | 0x80));
             buffer.put((byte) (value & 0xFF));
-            return;
+            return 16;
         }
 
         if ((value >>> 21) == 0) {
-            buffer.put((byte) ((value >> 16) | 192));
+            // 2 control bits
+            buffer.put((byte) ((value >> 16) | 0xC0));
             buffer.put((byte) ((value >> 8) & 0xFF));
             buffer.put((byte) (value & 0xFF));
-            return;
+            return 24;
         }
 
         if ((value >>> 28) == 0) {
-            buffer.put((byte) ((value >> 24) | 224));
+            // 3 control bits
+            buffer.put((byte) ((value >> 24) | 0xE0));
             buffer.put((byte) ((value >> 16) & 0xFF));
             buffer.put((byte) ((value >> 8) & 0xFF));
             buffer.put((byte) (value & 0xFF));
-            return;
+            return 32;
         }
 
-        buffer.put((byte) ((value >> 28) | 240));
+        // 4 control bits
+        buffer.put((byte) ((value >> 28) | 0xF0));
         buffer.put((byte) ((value >> 20) & 0xFF));
         buffer.put((byte) ((value >> 12) & 0xFF));
         buffer.put((byte) ((value >> 4) & 0xFF));
         buffer.put((byte) (value & 0xFF));
+        return 40;
+    }
+
+    /**
+     * Writes an unsigned (32 bit) integer to a byte new array encoded as ITF8. The sign bit is interpreted as a value bit.
+     *
+     * @param value the value to be written out
+     * @return the bytes holding ITF8 representation of the value
+     */
+    public static byte[] writeUnsignedITF8(final int value) {
+        final ByteBuffer buffer = ByteBuffer.allocate(10);
```

Comment:
I know this code was just moved from elsewhere, but is 10 meaningful here ? Although harmless, it seems like random over-allocation by half. Is there any way possible this can use more than the max possible # of bytes in an ITF8 encoding for an int, which is 5?. Can this use the new `MAX_BYTES` constant ?

Reply:
That makes sense to me.  I'll see how the tests respond.